src/drivers/CgreenDriver.cpp

```diff
@@ -0,0 +1,170 @@
+#include <cucumber-cpp/internal/drivers/CgreenDriver.hpp>
+
+#include <boost/bind.hpp>
+#include <boost/function.hpp>
+
+using namespace cgreen;
+
+namespace cucumber {
+namespace internal {
+
+
+bool CgreenStep::initialized = false;
+
+boost::function< void() > currentTestBody;
+
+static const char* const blacklist[] = {
+        "Running",
+        "Completed",
+        "Failure",
+        "currentTest",
+        "\n",
+        "/"
+};
+
+Ensure(currentTest) {
+    if(currentTestBody)
+    {
+        currentTestBody();
+    }
+}
+
+class CukeCgreenInterceptor {
+public:
+    const std::string getCgreenOutput();
+    static void clearCgreenOutput();
+    static void reset_reporter();
+    static int cgreenPrinter(const char* format, ...);
+    static TestSuite *cgreenSuite;
+    static TestReporter *cgreenReporter;
+    ~CukeCgreenInterceptor();
+
+private:
+    static std::string cgreenOutput;
+    static bool is_blacklisted(const char* input);
+};
+
+static CukeCgreenInterceptor cgreenInterceptor;
+
+std::string CukeCgreenInterceptor::cgreenOutput = std::string();
+
+TestSuite * CukeCgreenInterceptor::cgreenSuite = NULL;
+
+TestReporter * CukeCgreenInterceptor::cgreenReporter = NULL;
+
+CukeCgreenInterceptor::~CukeCgreenInterceptor() {
+    destroy_test_suite(cgreenSuite);
+    destroy_reporter(cgreenReporter);
+}
+
+const std::string CukeCgreenInterceptor::getCgreenOutput() {
+    return cgreenOutput;
+}
+
+
+void CukeCgreenInterceptor::clearCgreenOutput() {
+    cgreenOutput.clear();
+}
+
+void CukeCgreenInterceptor::reset_reporter() {
+    clearCgreenOutput();
+
+    cgreenReporter->passes = 0;
+    cgreenReporter->skips = 0;
+    cgreenReporter->failures = 0;
+    cgreenReporter->exceptions = 0;
+}
+
+int CukeCgreenInterceptor::cgreenPrinter(const char* format, ...) {
+    char buffer[10000] = { 0 };
+
+    va_list argPtr;
+
+    va_start(argPtr, format);
+
+    vsprintf(buffer, format, argPtr);
+
+    if(not is_blacklisted(buffer)) {
```

Comment:
Please just use the `!` operator instead of `not`. Very few C++ developers use the "word" form of the logical operators and it's not even available in C (the preprocessor doesn't count). Using the more common form (`!`) helps readability.